imap_processing/ialirt/l0/process_hit.py

```diff
@@ -0,0 +1,210 @@
+"""Functions to support HIT processing."""
+
+import logging
+from enum import Enum
+from typing import ClassVar
+
+import numpy as np
+import xarray as xr
+
+logger = logging.getLogger(__name__)
+
+
+class HITPrefixes(Enum):
+    """Create ENUM for each rate type."""
+
+    FAST_RATE_1: ClassVar[list[str]] = [
+        f"{prefix}_{i:02d}"
+        for i in range(15)
+        for prefix in ["L1A_TRIG", "IA_EVNT_TRIG", "A_EVNT_TRIG", "L3A_TRIG"]
+    ]
+    FAST_RATE_2: ClassVar[list[str]] = [
+        f"{prefix}_{i:02d}"
+        for i in range(15)
+        for prefix in ["L1B_TRIG", "IB_EVNT_TRIG", "B_EVNT_TRIG", "L3B_TRIG"]
+    ]
+    SLOW_RATE: ClassVar[list[str]] = [
+        "L1A",
+        "L2A",
+        "L3A",
+        "L1A0AHG",
+        "L1B0AHG",
+        "L1C0AHG",
+        "L4IAHG",
+        "L4OAHG",
+        "SLOW_RATE_08",
+        "SLOW_RATE_09",
+        "SLOW_RATE_10",
+        "L1A0BHG",
+        "L1B0BHG",
+        "L1C0BHG",
+        "L4IBHG",
+        "L4OBHG",
+        *[f"IALRT_RATE_{i}" for i in range(1, 21)],
```

Comment:
Are you going to need to decom the HIT science packet for IALRT RATE and ERATEs?

Reply:
Does the science packet have those values in it? The science packet won't ever make it over into the I-ALiRT stream, so this should be separate. But perhaps if you're planning on doing this on the science side as well (if they have duplicated that data into the science packet in addition to the I-ALiRT packet) you could use the I-ALiRT implementation here for that work.

Reply:
The science packet has fields called IALIRTRATES_00 to _19, so I'm not sure if that's needed for I-ALiRT. We're not using them for anything on our end, so I thought they'd be used by I-ALiRT. I imagine we won't be unpacking at a high enough frequency for I-ALiRT to use what we have, but we can share some code to help unpack those values from the science packet if those are needed.